Software/Python/I2C_mutex.py

```diff
@@ -1,27 +1,56 @@
 import fcntl
 import time
 
-DexterLockI2C_handle = None
+class Mutex(object):
 
+    DexterLockI2C_handle = None
+
+    def __init__(self, debug = False):
+        self.mutex_debug = debug
+
+    def acquire(self):
+        if self.mutex_debug:
+            print("I2C mutex acquire")
+        while self.DexterLockI2C_handle is not None:
+            time.sleep(0.001)
+        DexterLockI2C_handle = True
 
-def I2C_Mutex_Acquire():
-    global DexterLockI2C_handle
-    while DexterLockI2C_handle is not None:
-        time.sleep(0.001)
-    DexterLockI2C_handle = True # set to something other than None
-    DexterLockI2C_handle = open('/run/lock/DexterLockI2C', 'w')
-    acquired = False
-    while not acquired:
         try:
-            # lock
-            fcntl.lockf(DexterLockI2C_handle, fcntl.LOCK_EX | fcntl.LOCK_NB)
-            acquired = True
-        except IOError: # already locked by a different process
+            DexterLockI2C_handle = open('/run/lock/DexterLockI2C')
+        except:
+            pass
```

Comment:
what may cause this to fail? Wouldn't it be best to at least print the exception?

Reply:
No. There are 2 reasons:

1. Doing it so will lead to a very ugly console output, provided someone does use multiple threads/processes. Basically, the console will be filled with the said message.
2. Second of all, this is considered to be a bad practice. If we would want to do something about it, we could create a timeout on this method (most probably through another specialized class) and then end the method when the time runs out. Another option is using modules that are intended to be used for logging the output. Bottom line is: we should steer away from prints.

Also, this fails whenever the resource is locked by another thread/process - I wrote there a comment.

Reply:
Ok, you're right. All good then on this point